docs/azure.md

```diff
@@ -5,171 +5,305 @@
+    managedIdentity {
+        clientId = '<USER ASSIGNED MANAGED IDENTITY CLIENT ID>'
+    }
     storage {
-        // ...
-
-        fileShares {
-          rnaseqResources {
-                mountPath = "/mnt/mydata/myresources"
-            }
-        }
+        accountName = '<YOUR STORAGE ACCOUNT NAME>'
+    }
+    batch {
+        accountName = '<YOUR BATCH ACCOUNT NAME>'
+        location = '<YOUR BATCH ACCOUNT LOCATION>'
     }
 }
 ```
 
-The task can access the File share in `/mnt/mydata/myresources`. Note: The string `rnaseqResources` in the above config can be any name of your choice, and it does not affect the underlying mount.
-
-:::{warning}
-Azure File shares do not support authentication and management with Active Directory. The storage account key must be
-set in the configuration if a share is mounted.
+:::{note}
+Environment variable: `AZURE_MANAGED_IDENTITY_USER` can be used instead of the clientId setting
 :::
 
-(azure-batch)=
-
-## Azure Batch
+### Service Principals
 
-:::{tip}
-This section describes how to manually set up and use Nextflow with Azure Batch.
-You may be interested in using [Batch Forge](https://docs.seqera.io/platform/latest/compute-envs/azure-batch#compute-environment) in [Seqera Platform](https://seqera.io/platform/),
-which automatically creates the required Azure infrastructure for you with minimal intervention.
+:::{versionadded} 22.11.0-edge
 :::
 
-[Azure Batch](https://docs.microsoft.com/en-us/azure/batch/) is a managed computing service that allows the execution of containerised workloads in the Azure cloud infrastructure.
+[Service Principal](https://learn.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal) credentials can be used to access Azure Batch and Storage accounts. Similar to a Managed Identity, a Service Principal is an application identity that can have specific permissions and role-based access. Unlike Managed Identities, Service Principals require a secret key for authentication. This approach allows you to authenticate with Azure resources when operating outside of the Azure environment, such as from your local machine or a non-Azure cloud.
+
+```groovy
+azure {
+    activeDirectory {
+        servicePrincipalId = '<YOUR SERVICE PRINCIPAL CLIENT ID>'
+        servicePrincipalSecret = '<YOUR SERVICE PRINCIPAL CLIENT SECRET>'
+        tenantId = '<YOUR TENANT ID>'
+    }
+    storage {
+        accountName = '<YOUR STORAGE ACCOUNT NAME>'
+    }
+    batch {
+        accountName = '<YOUR BATCH ACCOUNT NAME>'
+        location = '<YOUR BATCH ACCOUNT LOCATION>'
+    }
+}
+```
 
-Nextflow provides built-in support for Azure Batch, allowing the seamless deployment of Nextflow pipelines in the cloud, in which tasks are offloaded as Batch jobs.
+:::{note}
+Environment variables can be used instead of config settings:
+- `AZURE_CLIENT_ID` - Service principal client ID
+- `AZURE_CLIENT_SECRET` - Service principal secret
+- `AZURE_TENANT_ID` - Tenant ID
```

Comment:
```suggestion
Environment variables can be used instead of config settings:

- `AZURE_CLIENT_ID`: Service principal client ID.
- `AZURE_CLIENT_SECRET`: Service principal secret.
- `AZURE_TENANT_ID`: Tenant ID.
```